index.js

```diff
@@ -161,178 +88,249 @@ function invalidationsAsPatches(invalidated, currentTree, nextTree) {
+            return createOperation();
+          } default: {
+            instrumentation.other++;
           }
-          return createOperation();
-        } default: {
-          instrumentation.other++;
         }
-      }
-    }).then(() => {
-      return queue(worker, pendingWork, this.concurrency);
-    }).then((result) => {
-      this._logger.info('applyPatches', 'duration:', timeSince(prevTime), JSON.stringify(instrumentation));
-      if (this.dependencies) {
-        this.processor.sealDependencies(this.dependencies);
-      }
-      this._needsReset = false;
-      return result;
+      }).then(() => {
+        return queue(worker, pendingWork, this.concurrency);
+      }).then((result) => {
+        this._logger.info('applyPatches', 'duration:', timeSince(prevTime), JSON.stringify(instrumentation));
+        if (this.dependencies) {
+          this.processor.sealDependencies(this.dependencies);
+        }
+        this._needsReset = false;
+        return result;
+      });
     });
-  });
-};
+  }
 
-Filter.prototype._handleFile = function(relativePath, srcDir, destDir, entry, outputPath, forceInvalidation, isChange, stats) {
-  stats.handleFile++;
+  _handleFile (relativePath, srcDir, destDir, entry, outputPath, forceInvalidation, isChange, stats) {
+    stats.handleFile++;
 
-  let handleFileStart = process.hrtime();
+    let handleFileStart = process.hrtime();
 
-  return new Promise(resolve => {
-    let result;
-    let srcPath = srcDir + '/' + relativePath;
+    return new Promise(resolve => {
+      let result;
+      let srcPath = srcDir + '/' + relativePath;
 
-    if (this.canProcessFile(relativePath, entry)) {
-      stats.processed++;
-      if (this._outputLinks[outputPath] === true) {
-        delete this._outputLinks[outputPath];
-        fs.unlinkSync(outputPath);
-      }
-      result = this.processAndCacheFile(srcDir, destDir, entry, forceInvalidation, isChange, stats);
-    } else {
-      stats.linked++;
-      if (isChange) {
-        fs.unlinkSync(outputPath);
+      if (this.canProcessFile(relativePath, entry)) {
+        stats.processed++;
+        if (this._outputLinks[outputPath] === true) {
+          delete this._outputLinks[outputPath];
+          fs.unlinkSync(outputPath);
+        }
+        result = this.processAndCacheFile(srcDir, destDir, entry, forceInvalidation, isChange, stats);
+      } else {
+        stats.linked++;
+        if (isChange) {
+          fs.unlinkSync(outputPath);
+        }
+        result = symlinkOrCopySync(srcPath, outputPath);
+        this._outputLinks[outputPath] = true;
       }
-      result = symlinkOrCopySync(srcPath, outputPath);
-      this._outputLinks[outputPath] = true;
-    }
-    resolve(result);
-  }).then(val => {
-    stats.handleFileTime += nanosecondsSince(handleFileStart);
-    return val;
-  }, err => {
-    stats.handleFileTime += nanosecondsSince(handleFileStart);
-    throw err;
-  });
-};
+      resolve(result);
+    }).then(val => {
```

Comment:
with async/await, we can simply do a syntax `finally` here, something like:

```js
async _handleFile(/* ... */) {
  try {
    // ...
    return /*..*/;
   } finally {
       stats.handleFileTime += nanosecondsSince(handleFileStart);
  }
}
```

Again, this aims to improve stack traces + debug tooling.